registry/registry.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"github.com/kubernetes-incubator/external-dns/endpoint"
+	"github.com/kubernetes-incubator/external-dns/plan"
+)
+
+// Registry is an interface which should enables ownership concept in external-dns
+// Record(zone string) returns ALL records registered with DNS provider (TODO: for multi-zone support return all records)
+// each entry includes owner information
+// ApplyChanges(zone string, changes *plan.Changes) propagates the changes to the DNS Provider API
+type Registry interface {
+	Records() []*endpoint.Endpoint
+	ApplyChanges(zone string, changes *plan.Changes) error
+}
```

Comment:
This interface is pretty similar to `Provider`. I'm thinking this could be just another provider, that wraps other providers to add the Ownership mechanism.

This could be interesting to try out with both the InMemory and some cloud DNS provider. Looking forward to this.